pkg/upgrade/drain_manager.go

```diff
@@ -83,12 +83,25 @@ func (m *DrainManagerImpl) ScheduleNodesDrain(ctx context.Context, drainConfig *
 		GracePeriodSeconds:  -1,
 		Timeout:             time.Duration(drainSpec.TimeoutSecond) * time.Second,
 		PodSelector:         drainSpec.PodSelector,
-		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
-			verbStr := "Deleted"
+		OnPodDeletionOrEvictionFinished: func(pod *corev1.Pod, usingEviction bool, err error) {
+			var verbStr string
 			if usingEviction {
-				verbStr = "Evicted"
+				if err != nil {
+					verbStr = "eviction failed"
+				} else {
+					verbStr = "evicted"
+				}
+			} else {
+				if err != nil {
+					verbStr = "deletion failed"
+				} else {
+					verbStr = "deleted"
+				}
+			}
+			if err != nil {
+				m.log.V(consts.LogLevelWarning).Info("pod deletion/eviction failed", "error", err)
 			}
-			m.log.V(consts.LogLevelInfo).Info(fmt.Sprintf("%s pod from Node %s/%s", verbStr, pod.Namespace, pod.Name))
+			m.log.V(consts.LogLevelInfo).Info(fmt.Sprintf("pod %s/%s %s\n", pod.Namespace, pod.Name, verbStr))
```

Comment:
An alternative could be

```go
			log := m.log.WithValues("pod", pod.Name, "namespace", pod.Namespace, "eviction", usingEviction)
			if err != nil {
				log.V(consts.LogLevelWarning).Info("Drain Pod failed", "error", err)
				return
			}
			log.V(consts.LogLevelInfo).Info("Drain Pod finished")
```

Reply:
Done, thanks for the review @tobiasgiese ! Please check now

Reply:
Looks good to me now. Unfortunately I am not able to approve